### ISSUE TITLE
Bai Wu strategy

### ISSUE THREAD

**Issue by jdmoorman** [jdmoorman/kaczmarz-algorithms#30]
Since it is common to compare adaptive rules against Bai and Wu's strategy, we should have an implementation of their selection strategy.

**Comment by jdmoorman** [comment]
The original method is described [here](https://epubs.siam.org/doi/pdf/10.1137/17M1137747?casa_token=qgsoQGz6UzMAAAAA:8NVx0IXS6541J-a2TDCEN9uZ_Dxn5egkCbQVEZXlA7MSm6t1cBc_nNbOixFaVi2TjwY7-IjrWrw) and a followup work which gives a slightly more general method can be found [here](https://www.sciencedirect.com/science/article/pii/S0893965918300739?casa_token=DYwgl2HKBKwAAAAA:HUOxBvc_MkzjwY51DgICHFSx21k81J1CNKPWE_uw0rWZqvVp7aJsU4xo_SIrsEMWSQhoiW7I2Q)!

Basically, sample rows with probability proportional to the normalized residual, but ignore rows whose normalized residual is below than some convex combination of the average and max normalized residual values. The parameter of the convex combination should be one half by default.
